### ISSUE TITLE
DEMO 2 seems broken

### ISSUE THREAD

**Issue by Kevin-Mattheus-Moerman** [mahtab-vafaee/LatticeWorks#2]
@mahtab-vafaee I currently get this error when trying to run [`DEMO_0002_SF_Multi_Morphology_TPMS.m`](https://github.com/mahtab-vafaee/LatticeWorks/blob/main/docs/DEMO_0002_SF_Multi_Morphology_TPMS.m). 

```
Arrays have incompatible sizes for this operation.

Error in triplyPeriodicMinimalSurface (line 68)
X=((X./abs(xMax-xMin)).*L);

Error in DEMO_0002_SF_Multi_Morphology_TPMS (line 62)
[~,~,~,S_A,X,Y,Z]=triplyPeriodicMinimalSurface(inputStruct_A);
```

Note that I've been editing some aspects which do not affect this error so the line numbers here are wrong.

**Comment by Kevin-Mattheus-Moerman** [comment]
Looks like I may have fixed this by editing GIBBON's `triplyPeriodicMinimalSurface` function: 
```matlab
if isscalar(L)
    L = L*ones(1,3); % Extend to 3D
end
```
To extend L to be 3D when only a scalar input is provided. 

Next I added: 
```matlab
%Scaling coordinates
X=((X./abs(xMax-xMin)).*L(1));
Y=((Y./abs(yMax-yMin)).*L(2));
Z=((Z./abs(zMax-zMin)).*L(3));
```
To use the L values for each direction. 

Next I also added the additional outputs that appear to be needed: 
```matlab
varargout{5}=X;
varargout{6}=Y;
varargout{7}=Z;
```

Do you think this addresses everything?

**Comment by mahtab-vafaee** [comment]
Thanks, Kevin. 
Yes, I guess that should be fine now, as I had edited the `triplyPeriodicMinimalSUrface` function in GIBBON to include non-cubic and 3D dimensions. 

The other change that I applied locally on this GIBBON function was adding a linear gradient factor, that was used as: 

```matlab
%% Gradient surface

switch length(L)
    case 1
        S=S.*(1-inputStruct.GF*Z/L);
    case 3
        S=S.*(1-inputStruct.GF*Z/L(1,3));
end
```
that initially `defaultInputStruct.GF=0`, but can be changed to apply levelset (volumetric) linear gradient, in different directions (in this example applies on `Z` direction). 

However, not sure if this change is used anywhere else in `LatticeWorks`.
